src/main/java/com/flightstats/hub/channel/ChannelResource.java

```diff
@@ -253,11 +258,11 @@ public EventOutput getEvents(@PathParam("channel") String channel, @HeaderParam(
 
     @DELETE
     public Response delete(@PathParam("channel") final String channelName) throws Exception {
-        ChannelConfig channelConfig = channelService.getChannelConfig(channelName, false);
-        if (channelConfig == null) {
+        Optional<ChannelConfig> optionalChannelConfig = channelService.getChannelConfig(channelName, false);
+        if (!optionalChannelConfig.isPresent()) {
             return notFound(channelName);
         }
-        if (HubProperties.isProtected() || channelConfig.isProtect()) {
+        if (HubProperties.isProtected() || optionalChannelConfig.get().isProtect()) {
```

Comment:
I was going to suggest chaining this stuff up with some `map`s and `filter`s, but it's probably a little gross.